TM1py/Services/CubeService.py

```diff
@@ -189,6 +189,40 @@ def get_dimension_names(self, cube_name: str, skip_sandbox_dimension: bool = Tru
             return dimension_names[1:]
         return dimension_names
 
+    def get_names_with_dimension(self, dimension_name: str, model_cubes_only: bool = False,
+                                 **kwargs) -> List[str]:
+        """ Ask TM1 Server for list of cube names that contain specific dimension
+
+        :param dimension_name: string, valid dimension name (case insensitive)
+        :param model_cubes_only: bool, True will filter result to model cubes only
+        """
+        cube_prefix = 'Cubes' if model_cubes_only == False else 'ModelCubes()'
+        url = format_url(
+            "/api/v1/{}?$select=Name&$filter=Dimensions/any(d: toupper(d/Name) eq toupper('{}'))",
```

Comment:
Does this filter take into account TM1's inherent space insensitivity?

Perhaps we need to include an odata string replace within the odata lambda.
Maybe check this function as a reference
https://github.com/cubewise-code/tm1py/blob/f19f58bcfdb65f2ae293762c50cb850d8e4a0bee/TM1py/Services/ObjectService.py#L25

Also, I think it's safer to use `tolower` rather than `toupper`. I learned the other day that the German `ß` is translated to `SS` in some libraries.
https://github.com/cubewise-code/mdxpy/issues/22

Reply:
I will need to test and confirm

Reply:
After testing, I think the functions are working as we'd intend them to, but perhaps I didn't test right.

I tested this by creating a dim "astest" and placing that dim in a cube and then attempted to search for the dim with both cube search functions looking for "as test" and it did not return any results. I then tried the reverse with a dim called "as test" and attempted to search based on dim "astest" and did not have any results returned.